### PR TITLE
Fixed per https://github.com/prancer-io/prancer-compliance-test/issues/342 and https://github.com/prancer-io/prancer-compliance-test/issues/343

### PR DESCRIPTION
**Pull request by rezoan**
Added rules for both Azure ARM and TF for https://github.com/prancer-io/prancer-compliance-test/issues/342

### CODE REVIEW

File: azure/iac/aks.rego [hunk 1]
```diff
@@ -665,3 +665,55 @@ aks_kub_dashboard_disabled_metadata := {
...
+
+aks_local_account_disabled = false {
+    azure_issue["aks_local_account_disabled"]
+}
+
+aks_local_account_disabled_err = "microsoft.containerservice/managedclusters property 'disableLocalAccounts' is missing from the resource" {
+    azure_attribute_absence["aks_local_account_disabled"]
+} else = "Azure Kubernetes Service Clusters currently dont have local authentication methods disabled" {
+    azure_issue["aks_local_account_disabled"]
+}
```
Comment (shahinshirvani) [inline azure/iac/aks.rego:706]: we should change the error massages like this:
```
aks_local_account_disabled_err = "Azure Kubernetes Service Clusters currently dont have local authentication methods disabled" {
    azure_issue["aks_local_account_disabled"]
} else = "microsoft.containerservice/managedclusters property 'disableLocalAccounts' is missing from the resource" {
    azure_attribute_absence["aks_local_account_disabled"]
}
```
priority of ```azure_issue``` more than ```azure_attribute_absence```
please update all other similar rules like this.

File: azure/terraform/master-compliance-test.json [hunk 2]
```diff
@@ -5751,6 +5820,145 @@
...
+                {
+                    "masterTestId": "TEST_STORAGE_ACCOUNT_15",
+                    "masterSnapshotId": [
+                        "TRF_TEMPLATE_SNAPSHOT"
+                    ],
+                    "type": "rego",
+                    "rule": "file(storageaccounts.rego)",
+                    "evals": [
+                        {
+                            "id": "PR-AZR-TRF-STR-021",
```
Comment (shahinshirvani) [inline azure/terraform/master-compliance-test.json:5902]: please change ```PR-AZR-TRF-STR-021``` to ```PR-AZR-TRF-STR-024``` to have same ID with IaC one

File: azure/terraform/master-compliance-test.json [hunk 3]
```diff
@@ -5751,6 +5820,145 @@
...
+                            "eval": "data.rule.storage_shared_access_key_disabled",
+                            "message": "data.rule.storage_shared_access_key_disabled_err",
+                            "remediationDescription": "In 'azurerm_storage_account' resource, set 'shared_access_key_enabled = false' to fix the issue. Visit <a href='https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#shared_access_key_enabled' target='_blank'>here</a> for details.",
+                            "remediationFunction": "PR_AZR_TRF_STR_021.py"
```
Comment (shahinshirvani) [inline azure/terraform/master-compliance-test.json:5906]: Change ```PR_AZR_TRF_STR_021``` to ```PR_AZR_TRF_STR_024```

File: azure/terraform/master-compliance-test.json [hunk 4]
```diff
@@ -5751,6 +5820,145 @@
...
+                {
+                    "masterTestId": "TEST_STORAGE_ACCOUNT_16",
+                    "masterSnapshotId": [
+                        "TRF_TEMPLATE_SNAPSHOT"
+                    ],
+                    "type": "rego",
+                    "rule": "file(storageaccounts.rego)",
+                    "evals": [
+                        {
+                            "id": "PR-AZR-TRF-STR-022",
```
Comment (shahinshirvani) [inline azure/terraform/master-compliance-test.json:5936]: please change ```PR-AZR-TRF-STR-022``` to ```PR-AZR-TRF-STR-023```

File: azure/terraform/master-compliance-test.json [hunk 5]
```diff
@@ -5751,6 +5820,145 @@
...
+                            "eval": "data.rule.storage_acl_usage_vnet",
+                            "message": "data.rule.storage_acl_usage_vnet_err",
+                            "remediationDescription": "In 'azurerm_storage_account_network_rules' resource or 'azurerm_storage_account's inner block 'network_rules', set 'default_action = Deny' and set id of target 'azurerm_subnet' into property 'virtual_network_subnet_ids' to fix the issue. Visit <a href='https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account_network_rules#virtual_network_subnet_ids' target='_blank'>here</a> for details.",
+                            "remediationFunction": "PR_AZR_TRF_STR_022.py"
```
Comment (shahinshirvani) [inline azure/terraform/master-compliance-test.json:5940]: please change ```PR_AZR_TRF_STR_022``` to ```PR_AZR_TRF_STR_023```

File: azure/terraform/storageaccounts.rego [hunk 6]
```diff
@@ -655,3 +670,269 @@ storage_account_latest_tls_configured_metadata := {
...
+    "Policy Title": "Storage account encryption scopes should use double encryption for data at rest",
+    "Policy Description": "Enable infrastructure encryption for encryption at rest of your storage account encryption scopes for added security. Infrastructure encryption ensures that your data is encrypted twice.",
+    "Resource Type": "azurerm_storage_account",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account"
+}
+
+
+#
+# PR-AZR-TRF-STR-021
```
Comment (shahinshirvani) [inline azure/terraform/storageaccounts.rego:802]: change 021 to 024

File: azure/terraform/storageaccounts.rego [hunk 7]
```diff
@@ -655,3 +670,269 @@ storage_account_latest_tls_configured_metadata := {
...
+}
+
+storage_shared_access_key_disabled_err = "azurerm_storage_account property 'shared_access_key_enabled' is missing from the resource" {
+    azure_attribute_absence["storage_shared_access_key_disabled"]
+} else = "Storage Accounts is currently not preventing shared key access" {
+    azure_issue["storage_shared_access_key_disabled"]
+}
+
+storage_shared_access_key_disabled_metadata := {
+    "Policy Code": "PR-AZR-TRF-STR-021",
```
Comment (shahinshirvani) [inline azure/terraform/storageaccounts.rego:840]: change 021 to 024

File: azure/terraform/storageaccounts.rego [hunk 8]
```diff
@@ -655,3 +670,269 @@ storage_account_latest_tls_configured_metadata := {
...
+    "Policy Title": "Storage accounts should prevent shared key access",
+    "Policy Description": "Audit requirement of Azure Active Directory (Azure AD) to authorize requests for your storage account. By default, requests can be authorized with either Azure Active Directory credentials, or by using the account access key for Shared Key authorization. Of these two types of authorization, Azure AD provides superior security and ease of use over Shared Key, and is recommended by Microsoft.",
+    "Resource Type": "azurerm_storage_account",
+    "Policy Help URL": "",
+    "Resource Help URL": "https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account"
+}
+
+
+#
+# PR-AZR-TRF-STR-022
```
Comment (shahinshirvani) [inline azure/terraform/storageaccounts.rego:853]: change 022 to 023

File: azure/terraform/storageaccounts.rego [hunk 9]
```diff
@@ -655,3 +670,269 @@ storage_account_latest_tls_configured_metadata := {
...
+    azure_attribute_absence["storage_acl_usage_vnet"]
+    azure_inner_attribute_absence["storage_acl_usage_vnet"]
+} else = "Storage Accounts currently not using virtual network service endpoint" {
+    lower(input.resources[_].type) == "azurerm_storage_account"
+    azure_issue["storage_acl_usage_vnet"]
+    azure_inner_issue["storage_acl_usage_vnet"]
+}
+
+storage_acl_usage_vnet_metadata := {
+    "Policy Code": "PR-AZR-TRF-STR-022",
```
Comment (shahinshirvani) [inline azure/terraform/storageaccounts.rego:928]: change 022 to 023

**Review by shahinshirvani — CHANGES_REQUESTED** [review]
can we support these two policies for terraform?
Storage account encryption scopes should use customer-managed keys to encrypt data at rest
Storage account encryption scopes should use double encryption for data at rest

File: azure/iac/aks.rego (hunk 1 above)
Comment (rezoan) [reply azure/iac/aks.rego:706]: @shahinshirvani i don't think that matters. as far as i know there is no priority. both will not fail at the same time.
any other reason that i dont know?

**Comment by rezoan** [comment]
> can we support these two policies for terraform? Storage account encryption scopes should use customer-managed keys to encrypt data at rest Storage account encryption scopes should use double encryption for data at rest

@shahinshirvani if you have missed my comment here https://github.com/prancer-io/prancer-compliance-test/issues/343#issuecomment-991630268
it has all the details.
The rule you are looking for `PR-AZR-TRF-STR-008` is already exist.

File: azure/terraform/master-compliance-test.json (hunk 2 above)
Comment (rezoan) [reply azure/terraform/master-compliance-test.json:5902]: done.

File: azure/terraform/master-compliance-test.json (hunk 3 above)
Comment (rezoan) [reply azure/terraform/master-compliance-test.json:5906]: Done.

File: azure/terraform/master-compliance-test.json (hunk 4 above)
Comment (rezoan) [reply azure/terraform/master-compliance-test.json:5936]: done.

File: azure/terraform/master-compliance-test.json (hunk 5 above)
Comment (rezoan) [reply azure/terraform/master-compliance-test.json:5940]: done.

File: azure/terraform/storageaccounts.rego (hunk 7 above)
Comment (rezoan) [reply azure/terraform/storageaccounts.rego:840]: done.

File: azure/terraform/storageaccounts.rego (hunk 8 above)
Comment (rezoan) [reply azure/terraform/storageaccounts.rego:853]: done.

File: azure/terraform/storageaccounts.rego (hunk 9 above)
Comment (rezoan) [reply azure/terraform/storageaccounts.rego:928]: done.

File: azure/terraform/storageaccounts.rego (hunk 6 above)
Comment (rezoan) [reply azure/terraform/storageaccounts.rego:802]: done.

File: azure/iac/aks.rego (hunk 1 above)
Comment (rezoan) [reply azure/iac/aks.rego:710]: @shahinshirvani got your point, its happening when we are not using `has_property `custom function.

**Review by farchide — APPROVED** [review]

**Review by shahinshirvani — APPROVED** [review]

**Pull request merged**